src/mjolnir/graphenhancer.cc

```diff
@@ -49,19 +45,93 @@ using namespace valhalla::mjolnir;
 
 namespace {
 
-// Geometry types for admin queries
-typedef boost::geometry::model::d2::point_xy<double> point_type;
-typedef boost::geometry::model::polygon<point_type> polygon_type;
-typedef boost::geometry::model::multi_polygon<polygon_type> multi_polygon_type;
-
 // Number of tries when determining not thru edges
 constexpr uint32_t kMaxNoThruTries = 256;
 
 // Radius (km) to use for density
 constexpr float kDensityRadius = 2.0f;
-constexpr float kDensityRadius2 = kDensityRadius * kDensityRadius;
 constexpr float kDensityLatDeg = (kDensityRadius * kMetersPerKm) / kMetersPerDegreeLat;
 
+struct DensityCellId {
+  // Density grid operates with cell_id derived from coordinate by shifting its bits by (32 -
+  // kGridLevel), meaning that bigger level produces smaller cells.
+  // - 17: 362m x 181m cell at 60 latitude, approx 7.5k cells to cover single tile
+  // - 18: 181m x 90m cell at 60 latitude, approx 30k cells to cover single tile
+  // - 19: 90m x 45m cell at 60 latitude, approx 120k cells to cover single tile
+  static constexpr uint32_t kGridLevel = 18; // good compromise between performance and accuracy
```

Comment:
The `18` was selected to keep density distribution as close as possible to the previous algorithm. `17` produces also good enough results with 0.25x CPU/RAM requirements for the density index

| level | Paris density map |
| --- | --- |
| 16 | <img width="1189" alt="image" src="https://github.com/user-attachments/assets/4e8bc5ed-8b4d-40c7-bdd2-d49d1480daa2" /> |
| 17 | <img width="1191" alt="image" src="https://github.com/user-attachments/assets/dfdcda4f-5f8a-4c0e-9aad-b3b2e8358ab6" /> |
| 18 | <img width="1190" alt="image" src="https://github.com/user-attachments/assets/0b502471-b3cf-4429-9f8d-f331b7224c25" /> |